### PR TITLE
fix(contrib/azure): use Azure premium storage

### PR DESCRIPTION
**Pull request by carmstrong**
Working with Microsoft, we suspect that cluster issues are due to poor disk performance.

/cc @sgoings

### CODE REVIEW

**Comment by carmstrong** [comment]
@helgi You're probably the best for a second LGTM

**Comment by helgi** [comment]
I'll have a look - Haven't worked on Azure since ARM came out. Looks like something to verify over pizza ;) Hold please

hold

File: contrib/azure/arm-template.json [hunk 1]
```diff
@@ -87,6 +87,13 @@
       "metadata": {
         "description": "Version of CoreOS to deploy."
       }
+    },
+    "storageAccountType": {
+      "type": "string",
+      "metadata": {
+        "description": "Storage account type. Can be Premium_LRS (recommended) or Standard_LRS"
+      },
+      "defaultValue": "Premium_LRS"
```
Comment (helgi) [inline contrib/azure/arm-template.json]: Can you add `allowedValues` (see https://azure.microsoft.com/en-us/documentation/articles/resource-group-authoring-templates/#parameters) so there is little room for mistakes.

Not sure how the view for this looks but would probably reduce the `description` part.

You may also want to add the rest of the ones from https://azure.microsoft.com/en-us/documentation/articles/resource-manager-template-storage/ - it's mostly just geo redundant and that like but won't hurt anything.

**Comment by helgi** [comment]
Not sure where we should drop it, perhaps the `azure` section of deis.io but linking to https://azure.microsoft.com/en-us/documentation/articles/storage-premium-storage-preview-portal/ with the Premium recommendation and a quick reasoning would be ace

File: contrib/azure/arm-template.json (hunk 1 above)
Comment (carmstrong) [inline contrib/azure/arm-template.json]: Good point on that - I thought of allowedValues after I pushed this, and wasn't sure it was worth doing. But I think it is.

**Comment by carmstrong** [comment]
@helgi Did you see my changes to the "Installing Deis on Azure" docs as part of this PR? Maybe it could use more info, but I did link to premium storage and the list of regions where it's enabled.

**Comment by helgi** [comment]
Nope, I was totally blind to that part! My bad :) Can I still get the `allowedValues` change in?

**Comment by colrack** [comment]
I'm testing a three node cluster with azure premium storage running in westeurope.
1 load balancer, 3 DS1 VMs. One day uptime no problem. I'll let you know.
However I tried a lot of possible configurations: LRS, GRS, ZRS, all kind of vm (A series, D series etc) and every time there was a ceph issue after some time (as reported here by many users).
Is it possible to start thinking about a stateless deis cluster using azure blob storage so we can use all kind of VMs without ceph storage issues?

**Comment by carmstrong** [comment]
> and every time there was a ceph issue after some time (as reported here by many users).

What sort of Ceph issue? Were the hosts going out of sync? If it's a matter of losing hosts after some time, I'm afraid that'll still require some manual intervention.

> Is it possible to start thinking about a stateless deis cluster using azure blob storage so we can use all kind of VMs without ceph storage issues?

If that's something you wanted to investigate and write documentation for, that'd be great! Azure blob store could work (as long as it's S3 compatible), and we'd still need an external Postgres instance.

**Comment by colrack** [comment]
So far when I tried to deploy to Azure I encountered storage related issues like ceph mon clock skew, #3969 "cls/rgw/cls_rgw.cc:1947: ERROR: rgw_obj_remove(): cls_cxx_remove returned -2", mon down I don't know why, hosts going out of sync and maybe other. Other issues maybe not storage related like #4557 "fatal: 'somerepo.git' does not appear to be a git repository", #4601 error from controller due to multiple instances of databases launched simultaneously...
Unfortunately Azure Blob Storage is not compatible with AWS S3. I just wondered how difficult would be to integrate it. (And yes we'd still need an external Postgres instance).

**Pull request merged**